seatunnel-connectors-v2/connector-paimon/src/main/java/org/apache/seatunnel/connectors/seatunnel/paimon/sink/PaimonSinkWriter.java

```diff
@@ -90,18 +93,33 @@ public PaimonSinkWriter(
             Table table,
             SeaTunnelRowType seaTunnelRowType,
             JobContext jobContext,
+            PaimonSinkConfig paimonSinkConfig,
             PaimonHadoopConfiguration paimonHadoopConfiguration) {
-        this.table = table;
+        this.table = (FileStoreTable) table;
+        CoreOptions.ChangelogProducer changelogProducer =
+                this.table.coreOptions().changelogProducer();
+        if (changelogProducer != paimonSinkConfig.getChangelogProducer()) {
+            log.warn(
+                    "configured 'changelog-producer' is not compatible with the table, will use the table's 'changelog-producer'");
```

Comment:
log.warn("Configured the props named 'changelog-producer'  which is not compatible with the options in table , so it will use the table's 'changelog-producer'");